langserver/symbol.go

```diff
@@ -66,14 +71,44 @@ func queryJoin(s, e string) string {
 	return strings.TrimSpace(s + " " + e)
 }
 
+func parseCaseSensitive(q string) (qu Query, fields []string) {
+	reflectVal := reflect.Indirect(reflect.ValueOf(qu)).Type()
+	FieldIndex := reflectVal.NumField() - 1
+	Field := reflectVal.Field(FieldIndex)
+	// Field: %+v: {Name:CaseSensitve PkgPath: Type:bool Tag: Offset:80 Index:[5] Anonymous:false}
+	FieldName := Field.Name
+	FieldNamePrefix := fmt.Sprintf("%s:", FieldName)
```

Comment:
hi,
i was just thinking if it were possible to have codegen tool parse to `Query` instead of writing the code to this...
a cursory look makes me think that it might be possible?

i'll get rid of the reflection.